src/neural/onnx/network_onnx.cc

```diff
@@ -413,6 +427,12 @@ std::unique_ptr<Network> MakeOnnxNetwork(const std::optional<WeightsFile>& w,
         fp16 ? WeightsToOnnxConverterOptions::DataType::kFloat16
              : WeightsToOnnxConverterOptions::DataType::kFloat32;
 
+    if (opts.GetOrDefault<bool>("bf16", false)) {
```

Comment:
I'd replace it with `datatype` = `f32`/`f16`/`bf16` parameter.
(see also this function https://github.com/LeelaChessZero/lc0/pull/1974/files#diff-4ede40c0bf9d771ac89c7f80c42de24aa6cb7bac22c215a95896b4ae28cf5709R1031 in my PR)